src/components/PackageDocs/Body.tsx

```diff
@@ -0,0 +1,68 @@
+import { LinkIcon } from "@chakra-ui/icons";
+import { Heading, As } from "@chakra-ui/react";
+import ChakraUIRenderer from "chakra-ui-markdown-renderer";
+import React from "react";
+import ReactMarkdown from "react-markdown";
+import { sanitize } from "../../util/sanitize-anchor";
+
+type HeadingResolverProps = {
+  level: number;
+  children: React.ReactNode;
+};
+
+function Headings({ level, children }: HeadingResolverProps) {
+  const size: string = ["2xl", "xl", "lg", "md", "sm", "xs"][level - 1];
+  const elem = `h${level}` as As<any>;
+
+  // Use DOMParser to look for data attribute for link ID
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(
+    React.Children.toArray(children).join(""),
+    "text/html"
+  );
+
+  const dataElement = doc.querySelector(
+    "span[data-heading-title][data-heading-id]"
+  ) as HTMLElement;
+  const title =
+    dataElement?.dataset.headingTitle ??
+    React.Children.toArray(children)
+      .reduce((accum: string, child): string => {
+        if (typeof child === "string") {
+          return `${accum}${child}`;
+        }
+        return accum;
+      }, "")
+      .trim();
+  const id = dataElement?.dataset.headingId ?? sanitize(title);
+
+  return (
+    <Heading my={4} level={level} as={elem} size={size}>
+      {level < 4 && (
```

Comment:
Why the restriction? I'm pretty sure we want to show all levels in the navigation...

Reply:
We can, it just feels extremely crowded and each construct has "initializer + methods + properties" and so there are a lot of duplicate titles in the left nav. I'd say perhaps once the menu is dynamic, IE collapses/expands, it may make it feels less crowded. But since there are like 6 levels it feels like that will also be hard to keep track of for the user.